src/test/java/org/prebid/server/settings/service/HttpPeriodicRefreshServiceTest.java

```diff
@@ -0,0 +1,166 @@
+package org.prebid.server.settings.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.VertxTest;
+import org.prebid.server.settings.CacheNotificationListener;
+import org.prebid.server.settings.proto.response.HttpRefreshResponse;
+import org.prebid.server.vertx.http.HttpClient;
+import org.prebid.server.vertx.http.model.HttpClientResponse;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.after;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+
+public class HttpPeriodicRefreshServiceTest extends VertxTest {
+
+
+    private static final String ENDPOINT_URL = "http://stored-requests.prebid.com";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private CacheNotificationListener cacheNotificationListener;
+
+    @Mock
+    private HttpClient httpClient;
+
+    private Vertx vertx;
+
+    private HttpPeriodicRefreshService httpPeriodicRefreshService;
+
+    private HttpRefreshResponse httpRefreshResponse;
+    private HttpClientResponse updateResponse;
+    private Map<String, String> expectedRequests;
+    private Map<String, String> expectedImps;
+
+    @Before
+    public void setUp() throws JsonProcessingException {
+        vertx = Vertx.vertx();
```

Comment:
Unfortunately, using mock causes issues in running test cases (probably it's vertx's behavior). I've checked other test classes and found out that we do use Vertx instance in @Before methods instead of mock.